app/lib/result.rb

```diff
@@ -2,7 +2,11 @@ class Result
   require 'uri'
   delegate :[], to: :@info
   def initialize(info)
-    @info = info
+    @info = if info.present?
+              info
+            else
+              {}
+            end
```

Comment:
Can also use a single line ternary to assign this:

`@info = info.present? ? info : {}`

Reply:
or easier yet 

```@info ||= {}```

Reply:
Done :) 
@info = info || {}